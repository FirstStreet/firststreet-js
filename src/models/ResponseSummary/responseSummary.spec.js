const BaseModel = require('./index');
const summaryMock = require('../../__mocks__/summary');

let summary = null;

describe('summaryMock', () => {
  beforeEach(() => {
    summary = new BaseModel(summaryMock);
  });

  it('should initialize a base model', () => {
    expect(summary).toMatchSnapshot();
  });

  it('should return a raw response', () => {
    expect(summary.raw).toBe(summaryMock);
  });

  it('should return the correct id', () => {
    expect(summary.FSID).toBe(summaryMock.FSID);
  });
  it('should return the correct results', () => {
    expect(summary.results).toBe(summaryMock.results);
  });
});
