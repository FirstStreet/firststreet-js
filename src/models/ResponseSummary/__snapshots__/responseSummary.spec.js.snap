// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summaryMock should initialize a base model 1`] = `
Response {
  "data": Object {
    "FSID": 100032470544,
    "results": Object {
      "firstFloodRisk": null,
      "floodRisks": null,
      "location": Object {
        "FSID": 100032470544,
        "city": Object {
          "FSID": 1047030,
          "name": "Middletown",
        ***REMOVED***,
        "distance": 0,
        "femaZone": null,
        "floorArea": null,
        "geometry": Object {
          "bbox": Object {
            "coordinates": Array [
              Array [
                Array [
                  -75.6456815055,
                  39.4418499121,
                ],
                Array [
                  -75.6451478218,
                  39.4418499121,
                ],
                Array [
                  -75.6451478218,
                  39.4421122702,
                ],
                Array [
                  -75.6456815055,
                  39.4421122702,
                ],
                Array [
                  -75.6456815055,
                  39.4418499121,
                ],
              ],
            ],
            "type": "Polygon",
          ***REMOVED***,
          "center": Object {
            "coordinates": Array [
              -75.6453718685,
              39.4419892115,
            ],
            "type": "Point",
          ***REMOVED***,
          "polygon": Object {
            "coordinates": Array [
              Array [
                Array [
                  Array [
                    -75.6451595133,
                    39.4418499121,
                  ],
                  Array [
                    -75.6451533033,
                    39.4418767065,
                  ],
                  Array [
                    -75.645149401,
                    39.4419037617,
                  ],
                  Array [
                    -75.6451478218,
                    39.441930957,
                  ],
                  Array [
                    -75.6451485745,
                    39.441958178,
                  ],
                  Array [
                    -75.645151654,
                    39.441985295,
                  ],
                  Array [
                    -75.6451570484,
                    39.4420121947,
                  ],
                  Array [
                    -75.6451647325,
                    39.4420387587,
                  ],
                  Array [
                    -75.6451746733,
                    39.4420648697,
                  ],
                  Array [
                    -75.6451868239,
                    39.4420904097,
                  ],
                  Array [
                    -75.6452018186,
                    39.4421122702,
                  ],
                  Array [
                    -75.6456586595,
                    39.4420479376,
                  ],
                  Array [
                    -75.6456815055,
                    39.4419626062,
                  ],
                  Array [
                    -75.6451595133,
                    39.4418499121,
                  ],
                ],
              ],
            ],
            "type": "MultiPolygon",
          ***REMOVED***,
        ***REMOVED***,
        "landUse": null,
        "lastUpdated": "2019-11-03T06:54:02.787801Z",
        "lotSize": null,
        "postdirection": "S",
        "predirection": null,
        "primaryNumber": "212",
        "state": "Delaware",
        "streetName": "APPOQUIN",
        "streetSuffix": "DR",
        "zipCode": 19709,
      ***REMOVED***,
    ***REMOVED***,
  ***REMOVED***,
***REMOVED***
`;
